hwy/ops/generic_ops-inl.h

```diff
@@ -462,6 +462,90 @@ HWY_API V IfNegativeThenNegOrUndefIfZero(V mask, V v) {
   return CopySign(v, Xor(mask, v));
 }
 
+// ------------------------------ SaturatedNeg
+
+#if (defined(HWY_NATIVE_SATURATED_NEG_8_16_32) == defined(HWY_TARGET_TOGGLE))
+#ifdef HWY_NATIVE_SATURATED_NEG_8_16_32
+#undef HWY_NATIVE_SATURATED_NEG_8_16_32
+#else
+#define HWY_NATIVE_SATURATED_NEG_8_16_32
+#endif
+
+template <class V, HWY_IF_T_SIZE_ONE_OF_V(V, (1 << 1) | (1 << 2)),
+          HWY_IF_SIGNED_V(V)>
+HWY_API V SaturatedNeg(V v) {
+  const DFromV<decltype(v)> d;
+
+#if HWY_TARGET == HWY_SCALAR || HWY_TARGET == HWY_EMU128
+  using T = TFromV<V>;
+  const RebindToUnsigned<decltype(d)> du;
+  const auto neg_v = Neg(v);
+  return Sub(
+      neg_v,
+      BitCast(d, ShiftRight<sizeof(T) * 8 - 1>(BitCast(du, And(v, neg_v)))));
+#else
+  return SaturatedSub(Zero(d), v);
+#endif
+}
+
+template <class V, HWY_IF_I32(TFromV<V>)>
+HWY_API V SaturatedNeg(V v) {
+  const DFromV<decltype(v)> d;
+
+#if HWY_TARGET == HWY_RVV ||                               \
+    (HWY_TARGET >= HWY_PPC10 && HWY_TARGET <= HWY_PPC8) || \
+    (HWY_TARGET >= HWY_SVE2_128 && HWY_TARGET <= HWY_NEON_WITHOUT_AES)
+  // RVV/NEON/SVE/PPC have native I32 SaturatedSub instructions
+  return SaturatedSub(Zero(d), v);
+#elif HWY_TARGET == HWY_SCALAR || HWY_TARGET == HWY_EMU128
+  const RebindToUnsigned<decltype(d)> du;
+  const auto neg_v = Neg(v);
+  return Sub(neg_v, BitCast(d, ShiftRight<31>(BitCast(du, And(v, neg_v)))));
+#else
+  return Sub(Not(v), VecFromMask(d, Gt(v, Set(d, LimitsMin<int32_t>()))));
```

Comment:
Oh nice, this is surprisingly efficient 👍  So 0x8..0 1's complement is 0x7F..F and otherwise to get 2's complement, we add one. A comment like this would help future readers :)